### PR TITLE
feat: implement "for own" in a statement context

### LINKED ISSUE #157: Add support for `for own` loops
```coffee
for own a of b
  a
```

### PR DESCRIPTION
**Pull request by alangpierce**
We can just use a JS for-of loop on `Object.keys`, which is defined to iterate
through the own properties in the same order that JS for-in would.

Closes #157.

### CODE REVIEW

**Comment by madskonradsen** [comment]
Love the prettier syntax <3

File: test/for_test.js [hunk 1]
```diff
@@ -475,67 +475,58 @@ describe('for loops', () => {
...
     `);
   });
 
-  it.skip('handles single-line `for own`', () => {
+  it('handles single-line `for own`', () => {
     check(`
       for own a of b then a
     `, `
-      for (let a in b) { if (Object.prototype.hasOwnProperty.call(b, a)) { a; } }
+      for (let a of Object.keys(b)) { a; }
```
Comment (eventualbuddha) [inline test/for_test.js]: We should also test that unsafe-to-repeat lists work like this.

Comment (eventualbuddha) [inline test/for_test.js]: That is, when accessing the value at each key.

Comment (alangpierce) [inline test/for_test.js]: Good call, I thought that would just work like it works for normal `for of`, but it breaks here. I'll fix and add a test.

**Comment by alangpierce** [comment]
Ok, I just updated the commit to handle non-repeatable targets where both the key and value are bound. I ended up just reworking the code to use the same code and approach as `ForInPatcher`, which also makes some other cases look nicer.

Another thought here is to loop over `Object.entries`, so the expression doesn't need to be repeated in the first place, but for now I just stuck with the existing approach.

**Comment by eventualbuddha** [comment]
Thanks!

**Pull request merged**